File: csharp/Practice01/src/MyClient/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;

namespace MyClient
{
    public class Program
    {
        public static void Main(string[] args)
        {
            var sender = new MySender("jeffz://server:12345");

            try
            {
                sender.AddQuery(1);
                sender.AddQuery(2);
                sender.AddQuery(3);
            }
            catch
            {
                Console.WriteLine("Error occurred when add query.");
                Environment.Exit(1);
            }

            new Thread(() => ReceiveData(sender.Receiver)).Start();
        }

        private static void ReceiveData(MyReceiver receiver)
        {
            try
            {
                while (true)
                {
                    var data = receiver.Receive();
                    if (data == null)
                    {
                        Console.WriteLine("Closed");
                        break;
                    }
                    else
                    {
                        Console.WriteLine(data);
                    }
                }
            }
            catch (MyClientException)
            {
                Console.WriteLine("Error occurred when receive data.");
            }
        }
    }
}

File: java/Practice01/src/myClient/Program.java
package myClient;

public class Program {

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		
		final MySender sender = new MySender("jeffz://server:12345");

        try {
            sender.addQuery(1);
            sender.addQuery(2);
            sender.addQuery(3);
        }
        catch (MyClientException ex) {
            System.out.println("Error occurred when add query.");
            System.exit(1);
        }

        new Thread(new Runnable() {

			@Override
			public void run() {
				receiveData(sender.getReceiver());
			}
			
        }).start();
	}

	private static void receiveData(MyReceiver receiver) {
		try {
            while (true) {
                MyData data = receiver.receive();
                if (data == null) {
                    System.out.println("Closed");
                    break;
                }
                else {
                    System.out.println(data);
                }
            }
        }
        catch (MyClientException ex) {
            System.out.println("Error occurred when receive data.");
        }
	}
}
